client/verta/tests/http_requests/test_client.py

```diff
@@ -0,0 +1,183 @@
+        hosts = [
+            "www.google.com",
+        ]
+
+        for host in hosts:
+            # http by default
+            try:
+                verta.Client(host, max_retries=0)
+            except requests.HTTPError as e:
+                assert e.request.url.split(':', 1)[0] == "http"
+            else:
+                raise RuntimeError("faulty test; expected error")
+
+            # http if provided
+            try:
+                verta.Client("http://{}".format(host), max_retries=0)
+            except requests.HTTPError as e:
+                assert e.request.url.split(':', 1)[0] == "http"
+            else:
+                raise RuntimeError("faulty test; expected error")
+
+            # https if provided
+            try:
+                verta.Client("https://{}".format(host), max_retries=0)
+            except requests.HTTPError as e:
+                assert e.request.url.split(':', 1)[0] == "https"
+            else:
+                raise RuntimeError("faulty test; expected error")
+
+    @pytest.mark.not_oss
+    def test_wrong_credentials(self):
+        EMAIL_KEY, DEV_KEY_KEY = "VERTA_EMAIL", "VERTA_DEV_KEY"
+        old_email, old_dev_key = os.environ[EMAIL_KEY], os.environ[DEV_KEY_KEY]
+
+        try:
+            os.environ[EMAIL_KEY] = "abc@email.com"
+            os.environ[DEV_KEY_KEY] = "def"
+
+            with pytest.raises(requests.exceptions.HTTPError) as excinfo:
+                verta.Client()
+
+            excinfo_value = str(excinfo.value).strip()
+            assert "401 Client Error" in excinfo_value
+            assert "authentication failed; please check `VERTA_EMAIL` and `VERTA_DEV_KEY`" in excinfo_value
+        finally:
+            os.environ[EMAIL_KEY], os.environ[DEV_KEY_KEY] = old_email, old_dev_key
+
+    @pytest.mark.not_oss
+    def test_ca_bundle_env_var(self, client):
+        """"Verify make_request() honors REQUESTS_CA_BUNDLE env var."""
+        REQUESTS_CA_BUNDLE_ENV_VAR = "REQUESTS_CA_BUNDLE"
+        good_ca_bundle_path = os.environ.get(REQUESTS_CA_BUNDLE_ENV_VAR)
+        conn = client._conn
+
+        url = "{}://{}/".format(conn.scheme, conn.socket)
+
+        # as a control, make sure request works
+        response = _utils.make_request("GET", url, conn)
+        conn.must_response(response)
+
+        bad_ca_bundle_path = "foo"
+        msg_match = (
+            "^Could not find a suitable TLS CA certificate bundle,"
+            " invalid path: {}$".format(bad_ca_bundle_path)
+        )
+        error_type = IOError if six.PY2 else OSError
+        try:
+            os.environ[REQUESTS_CA_BUNDLE_ENV_VAR] = bad_ca_bundle_path
+
+            with pytest.raises(error_type, match=msg_match):
+                _utils.make_request("GET", url, client._conn)
+        finally:
+            if good_ca_bundle_path:
+                os.environ[REQUESTS_CA_BUNDLE_ENV_VAR] = good_ca_bundle_path
+            else:
+                del os.environ[REQUESTS_CA_BUNDLE_ENV_VAR]
```

Comment:
This is the only new test in this file; everything else came from `test_entities.py`